### ISSUE TITLE
Add Sequence Data Loader for LSTM Training

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#29]
**Goal**: Enable LSTM betting strategies by implementing data infrastructure to query and prepare time-series odds sequences. The existing `SequenceFeatureExtractor` handles feature engineering, but we need utilities to fetch historical line movement data and convert it into the 3D tensor format required for LSTM training and prediction.

**Complexity**: 🟡 Medium

**Location**:
- `packages/odds-analytics/odds_analytics/sequence_data_loader.py` (new module)
- `tests/unit/test_sequence_data_loader.py` (new test file)

**Implementation Approach**:

Create a new module with these core functions:

```python
async def load_sequences_for_event(
    event: BacktestEvent,
    session: AsyncSession,
    market: str = "h2h",
    outcome: str | None = None,
    lookback_hours: int = 72,
) -> list[list[Odds]]:
    """
    Query all odds snapshots for an event and group by timestamp.
    
    Returns:
        List of snapshots ordered chronologically (oldest to newest)
        Each snapshot is a list of Odds records at the same timestamp
    """

async def prepare_lstm_training_data(
    events: list[BacktestEvent],
    session: AsyncSession,
    extractor: SequenceFeatureExtractor,
    market: str = "h2h",
    outcome_selector: Callable[[BacktestEvent], list[str]] = None,
) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
    """
    Prepare 3D training data for LSTM from multiple events.
    
    Args:
        events: List of events with final scores
        session: Database session
        extractor: Configured SequenceFeatureExtractor
        market: Market type (h2h, spreads, totals)
        outcome_selector: Function to determine which outcomes to extract per event
                         (default: both home and away)
    
    Returns:
        X: Features (n_samples, timesteps, n_features)
        y: Labels (n_samples,) - binary win/loss
        masks: Attention masks (n_samples, timesteps) - True where data exists
    """
```

**Constraints**:
- Must use async/await pattern (all database operations are async)
- Must use `OddsReader.get_line_movement()` or `OddsReader.get_snapshots_for_event()` for queries
- Must handle variable-length sequences (not all events have same number of snapshots)
- Must preserve chronological order (critical for LSTM training)
- Cannot assume all bookmakers present in all snapshots (handle missing data gracefully)

**Scope**:
- Implement `load_sequences_for_event()` to query and group odds by timestamp
- Implement `prepare_lstm_training_data()` to batch-process multiple events
- Create labels from event outcomes (home win = 1, away win = 0 for away team predictions)
- Handle edge cases: no snapshots, partial data, missing bookmakers
- Write unit tests with synthetic data covering various scenarios
- Write integration test that queries real database (if test data exists)

**Success Criteria**:
- [ ] `load_sequences_for_event()` returns chronologically ordered snapshots
- [ ] `prepare_lstm_training_data()` produces 3D arrays with correct shapes
- [ ] Labels correctly map to event outcomes (won=1, lost=0)
- [ ] Attention masks correctly indicate valid vs padded timesteps
- [ ] Unit tests pass for: normal case, no snapshots, partial data, single snapshot
- [ ] Integration test passes against test database (using existing test fixtures)
- [ ] Type hints use modern syntax (`str | None`, not `Optional[str]`)
- [ ] No ruff or type errors introduced

**Related Files**:
- `packages/odds-lambda/odds_lambda/storage/readers.py` - Use `OddsReader.get_snapshots_for_event()` or `get_line_movement()` for querying
- `packages/odds-analytics/odds_analytics/feature_extraction.py` - `SequenceFeatureExtractor` is the consumer of this data loader
- `packages/odds-core/odds_core/models.py` - `Odds` and `Event` model definitions
- `tests/unit/test_feature_extraction.py` - Test patterns for feature extraction (lines 600-891 cover sequence features)

**Notes**:
- The `SequenceFeatureExtractor.extract_features()` expects `list[list[Odds]]` where each inner list is a snapshot
- Snapshots must be grouped by `odds_timestamp` field, not by database insertion time
- For training, you'll need to query events in a date range and process each one
- Consider batching database queries (don't query each event individually) for performance
- Look-ahead bias prevention: Training data should only use snapshots before game starts
